src/components/Questionnaire/QuestionTypes/AllergyQuestion.tsx

```diff
@@ -422,7 +420,16 @@ export function AllergyQuestion({
                   <AllergyTableRow
                     key={index}
                     allergy={allergy}
-                    disabled={disabled}
+                    disabled={
+                      disabled ||
+                      patientAllergies?.results
```

Comment:
create a utility fn you can use across all the files. Other than that, seems fine. Nihal will check if the behavior is what we want in mobile.